src/index.js

```diff
@@ -1,20 +1,75 @@
 'use strict';
 
-var Class = require('class.extend');
+var _ = require('underscore'),
+    Class = require('class.extend');
 
 module.exports = Class.extend({
 
    init: function(serverless, opts) {
       this._serverless = serverless;
       this._provider = serverless ? serverless.getProvider('aws') : null;
       this._opts = opts;
+      this._service = serverless.service;
       this._custom = serverless.service ? serverless.service.custom : null;
 
       if (!this._provider) {
          throw new Error('This plugin must be used with AWS');
       }
 
-      throw new Error('This plugin\'s functionality has not been implemented');
+      this.hooks = {
+         'deploy:finalize': this._addTagsToStages.bind(this),
+      };
+   },
+
+   _addTagsToStages: function() {
+      var self = this,
+          stackTags = this._getStackTags();
+
+      return this._getAPIGatewayStages()
+         .then(function(stages) {
+            return _.map(stages, function(stage) {
+               return self._tagAPIGatewayStage(stage.arn, stackTags);
+            });
+         });
+   },
+
+   _tagAPIGatewayStage: function(arn, tags) {
+      var params;
+
+      params = {
+         resourceArn: arn,
+         tags: tags,
+      };
+
+      return this._provider.request('APIGateway', 'tagResource', params);
+   },
+
+   _getStackTags: function() {
+      return _.extend({
+         STAGE: this._service.provider.stage, // The provider's stackTags does not include the stage
```

Comment:
Same question here that I had on https://github.com/silvermine/serverless-plugin-cloudfront-distribution-tags/pull/1 ... although since in this PR, we're using the `deploy:finalize` hook, it seems like the answer is "Serverless never adds STAGE to the stack tags"... If that's the case, 🤷‍♂️ what can we do. The only thing that might be later is `after:deploy:finalize`, but I'm guessing that won't matter.

Reply:
Yes, `STAGE` never is added to the provider's stackTags. For posterity, note [this response](https://github.com/silvermine/serverless-plugin-cloudfront-distribution-tags/pull/1#discussion_r218845252) on silvermine/serverless-plugin-cloudfront-distribution-tags#1.